m4a_hq_mixer.s

```diff
@@ -1115,20 +1045,21 @@ C_downsampler_loop:
 .endif
     SUBS    R8, #2
     BGT     C_downsampler_loop
-
+    /* Switch to Thumb */
     ADR     R0, (C_downsampler_return+1)
     BX      R0
 
     .pool
 
     .align  1
     .thumb
+    .thumb_func
```

Comment:
what does this do here?

Reply:
This merely tells the assembler to mark the label as a thumb function so it shows up in disassembles as thumb, and if it were to be externally referenced as a global it would automatically set the thumb bit.

Reply:
For me it shows up in the disassembly without it just fine. Furthermore this should not be called externally. Yes yes, I know that I should probably convert the labels to something more common like `.L` type of label names.